src/components/progress-circle/progress-circle.scss

```diff
@@ -0,0 +1,97 @@
+@import "variables";
+
+@import "default-theme";
+
+
+$circle-ease-in-out : cubic-bezier(0.35, 0, 0.25, 1) !default;
+$circle-duration : 5.25s !default;
+$value-change-duration : $circle-duration * 0.25 !default;
+$constant-rotate : $circle-duration * (5 / 9) !default;
+$sporadic-rotate-duration : $circle-duration !default;
+$circle-stroke-width: 10px !default;
+$circle-radius: 40px !default;
+$circle-circumference: $pi * $circle-radius * 2 !default;
+$circle-center-point: 50px !default;
+$default-circle-size : 100px !default;
+
+
+md-progress-circle,
+md-spinner {
+  display: block;
+  /** Height and width are provided for md-progress-circle to act as a default.
+      The height and width are expected to be overwritten by application css. */
+  height: $default-circle-size;
+  width: $default-circle-size;
+
+  /** SVG's viewBox is defined as 0 0 100 100, this means that all SVG children will placed
+      based on a 100px by 100px box.
+
+      The SVG and Circle dimensions/location:
+        SVG
+          Height: 100px
+          Width: 100px
+        Circle
+          Radius: 40px
+          Circumference: 251.3274px
+          Center x: 50px
+          Center y: 50px
+  */
+  svg {
+    height: 100%;
+    width: 100%;
+  }
+
+  circle {
+    cx: $circle-center-point;
+    cy: $circle-center-point;
+    fill: transparent;
+    r: $circle-radius;
+    stroke: md-color($md-primary, 100);
+    /** Stroke width of 10px defines stroke as 10% of the viewBox */
+    stroke-width: $circle-stroke-width;
+    /** SVG circle rotations begin rotated 90deg clockwise from the circle's center top */
+    transform: rotate(-90deg);
+    transform-origin: center;
+    transition: stroke-dashoffset 0.225s linear;
+    /** The dash array of the circle is defined as the circumference of the circle. */
+    stroke-dasharray: $circle-circumference;
+    /** The stroke dashoffset is used to "fill" the circle, 0px represents an full circle,
+        while the circles full circumference represents an empty circle. */
+    stroke-dashoffset: 0px;
+  }
+
+  &.md-accent circle {
+    stroke: md-color($md-accent, 100);
+  }
+
+  &.md-warn circle {
+    stroke: md-color($md-warn, 100);
+  }
+
+  &[mode="indeterminate"] circle {
+    animation: sporadic-rotate $sporadic-rotate-duration $circle-ease-in-out infinite,
+               constant-rotate $constant-rotate linear infinite,
+               value-change $value-change-duration $circle-ease-in-out infinite;
+    transition: none;
+  }
+}
+
+/** Animations for indeterminate mode */
+@keyframes constant-rotate {
```

Comment:
Can you make these names more specific (like "md-spinner-constant-rotate") and also add a description of what each animation does specifically?